social.md

```diff
@@ -11,134 +11,273 @@ support:
-
-In this situation, the user leaves your login page by redirect and is taken to
-the provider for authentication.  Once authentication is complete, the user is
-redirected back to a "callback" URI on your website.  This callback will provide
-an access token or access code as a query parameter.  Your server uses a
-confidential keypair of the provider to validate the token/code, then retrieves
-the account data of the authenticated user.
-
-In order to support this workflow, the framework integration must:
+Coming Soon:
 
-* Parse the provider configuration of the specified Stormpath application (see
-  next section).
-
-* Provide callback handlers for this workflow.  See "Implementing Page-Based
-  Workflows" in this document.
-
-### AJAX-based Workflow
-
-In this situation the user does not leave your login page.  Instead a popup
-window is created, and the user authenticates with the provider in that window.
-The popup window is created by a JavaScript API in the browser, and when the
-user is done with the popup window (either by authenticating or rejecting the
-prompt) the JavaScript API will call back to your JavaScript application.  If
-the user has authenticated you will be provided with a token/code, which you
-must send to your server and validate with your confidential provider
-credentials.  At this point the workflow is identical to the page-base redirect
-workflow.
-
-In order to support this worklow, the framework integration must do the
-following:
-
-  * Parse the provider configuration of the specified Stormpath application (see
-    next section).
-
-  * Accept the access token or code via the login endpoint that this framework
-    provides, see [login][] for implementation spec.
-
-  * Render buttons for social login, on the registration and login views (or
-    view model), see [login][] and [registration][].
+* Twitter
+* Generic OAuth
 
+## Background: Social Login Workflows
 
-## Parsing Provider Configuration
+Social providers typically implement two authentication workflows: [page-based redirect](#page-based) workflow and [ajax-based](#ajax-based) worflow.
+
+The framework integration interacts with Stormpath via the Stormpath Client API in a uniform way - regardless of the social provider.
+
+The Stormpath Client API interacts with the Stormpath REST API which in turn, interacts directly with the social provider using its page-based redirect workflow.
+
+All of these details are hidden from the framework integration. The framework integration can implement an ajax-based workflow or a page-based workflow as it sees fit.
+
+There are two primary responsibilties of the framework integration:
+
+1. Obtain a URL to interact with the Client API
+2. Validate and parse a Stormpath JWT assertion response
+
+## Obtain a URL to interact with the Client API
+
+The Social Provider login flow starts with a call to the `/authorize` endpoint of the Client API. There are two ways to obtain the proper URL:
```

Comment:
Let's say you've customized the login model in your integration. Maybe you have the placeholder text for login set to a different value than the default. This would not automatically be reflected by the client API. Therefore, I thought it was important to specify that the integration could construct the `/authorize` url and still use their own login model.